pkg/reconciler/workload/resource/resource_reconcile_test.go

```diff
@@ -0,0 +1,187 @@
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync",
+			}),
+			obj: object(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-3": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync",
+			}),
+		},
+		{name: "hard delete",
+			ns: namespace(nil, nil),
+			obj: object(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-3": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync", // removed hard
+			}),
+			wantLabelPatch: map[string]interface{}{
+				"state.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+			wantAnnotationPatch: map[string]interface{}{
+				"deletion.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+		},
+		{name: "existing deletion, hard delete",
+			ns: namespace(nil, nil),
+			obj: object(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-3": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync",
+			}),
+			wantLabelPatch: map[string]interface{}{
+				"state.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+			wantAnnotationPatch: map[string]interface{}{
+				"deletion.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+		},
+		{name: "existing deletion, rescheduled",
+			ns: namespace(nil, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync",
+			}),
+			obj: object(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-3": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync",
+			}),
+		},
+		{name: "multiple, added and removed",
+			ns: namespace(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-4": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-1": "Sync",
+				"state.internal.workloads.kcp.dev/cluster-2": "Sync",
+				"state.internal.workloads.kcp.dev/cluster-4": "Sync", // deleting
+			}),
+			obj: object(map[string]string{
+				"deletion.internal.workloads.kcp.dev/cluster-3": "2002-10-02T10:00:00-05:00",
+			}, map[string]string{
+				"state.internal.workloads.kcp.dev/cluster-2": "Sync",
+				"state.internal.workloads.kcp.dev/cluster-3": "Sync", // removed hard
+				"state.internal.workloads.kcp.dev/cluster-4": "Sync",
+			}),
+			wantLabelPatch: map[string]interface{}{
+				"state.internal.workloads.kcp.dev/cluster-1": "Sync",
+				"state.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+			wantAnnotationPatch: map[string]interface{}{
+				"deletion.internal.workloads.kcp.dev/cluster-4": "2002-10-02T10:00:00-05:00",
+				"deletion.internal.workloads.kcp.dev/cluster-3": nil,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotAnnotationPatch, gotLabelPatch := computePlacement(tt.ns, tt.obj)
+			if !equality.Semantic.DeepEqual(gotAnnotationPatch, tt.wantAnnotationPatch) {
```

Comment:
Can we please always use the following on larger objects

```go
if diff := cmp.Diff(gotAnnotationPatch, tt.wantAnnotationPatch); diff != "" {
    t.Errorf("incorrect annotation patch: %s", diff)
}
```

Reply:
❤️